File: wx-share-java/src/main/java/com/sch/share/utils/ClipboardUtil.java
package com.sch.share.utils;

import android.content.ClipData;
import android.content.ClipboardManager;
import android.content.Context;

/**
 * Created by StoneHui on 2018/12/12.
 * <p>
 * 剪切板工具类。
 */
public class ClipboardUtil {

    /**
     * 复制内容到剪切板。
     */
    public static void setPrimaryClip(Context context, CharSequence label, CharSequence text) {
        ClipboardManager manager = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        if (manager != null) {
            manager.setPrimaryClip(ClipData.newPlainText(label, text));
        }
    }

    /**
     * 获取剪切板的内容。
     */
    public static CharSequence getPrimaryClip(Context context) {
        ClipboardManager manager = (ClipboardManager) context.getSystemService(Context.CLIPBOARD_SERVICE);
        if (manager == null) {
            return "";
        }
        ClipData data = manager.getPrimaryClip();
        if (data == null || data.getItemCount() <= 0) {
            return "";
        }
        ClipData.Item item = data.getItemAt(0);
        return item != null ? item.getText() : "";
    }

}

File: wx-share/src/main/java/com/sch/share/utils/ClipboardUtil.kt
package com.sch.share.utils

import android.content.ClipData
import android.content.ClipboardManager
import android.content.Context

/**
 * Created by StoneHui on 2018/10/30.
 * <p>
 * 剪切板工具类。
 */
object ClipboardUtil {

    /**
     * 复制内容到剪切板。
     */
    fun setPrimaryClip(context: Context, label: CharSequence, text: CharSequence) {
        (context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager)
                .primaryClip = ClipData.newPlainText(label, text)
    }

    /**
     * 获取剪切板的内容。
     */
    fun getPrimaryClip(context: Context): CharSequence {
        return (context.getSystemService(Context.CLIPBOARD_SERVICE) as ClipboardManager)
                .primaryClip?.getItemAt(0)?.text ?: ""
    }

}